torch/csrc/jit/passes/tensorexpr_fuser.cpp

```diff
@@ -393,6 +451,131 @@ class TensorExprFuser {
   }
 #undef REQ
 
+  void fillTypeInfoMap(Block* block) {
+    for (Node* n : block->nodes()) {
+      if (n->kind() == prim::profile && n->outputs().size() == 1) {
+        if (auto tensor_ty = n->ty(attr::profiled_type)->cast<TensorType>()) {
+          GRAPH_DEBUG(
+              "TYPEINFO ADD: %", n->input()->debugName(), " --> ", *tensor_ty);
+          typeinfo_map_[n->input()] = tensor_ty;
+        }
+      }
+
+      for (Block* b : n->blocks()) {
+        fillTypeInfoMap(b);
+      }
+    }
+  }
+  void printTypeInfoMap() {
+    GRAPH_DEBUG("Typeinfo map:");
+    for (const auto& kv : typeinfo_map_) {
+      GRAPH_DEBUG("%", kv.first->debugName(), " --> ", *kv.second);
+    }
+  }
+
+  void removeProfilingNodes(Block* b) {
```

Comment:
nitpick: I'm not sure `removeProfilingNodes` belongs in this file. `profiling_record.h`, `profiling_graph_executor_impl.h` might be a better choice.

Reply:
Yeah, it probably doesn't belong here. I plan to do a spring cleaning of PE-related passes after our changes land and stabilize a little, and as a part of it I'll move this function to a better place.